engine/table/src/main/java/io/deephaven/engine/table/impl/locations/local/URIStreamKeyValuePartitionLayout.java

```diff
@@ -0,0 +1,143 @@
+        final KeyValuePartitionLayout.LocationTableBuilder locationTableBuilder = locationTableBuilderFactory.get();
+        final Deque<URI> targetURIs = new ArrayDeque<>();
+        final Set<String> takenNames = new HashSet<>();
+        final List<String> partitionKeys = new ArrayList<>();
+        final boolean[] registered = {false}; // Hack to make the variable final
+        uriStream.forEachOrdered(uri -> {
+            final Collection<String> partitionValues = new ArrayList<>();
+            final URI relativePath = tableRootDirectory.relativize(uri);
+            getPartitions(relativePath, partitionKeys, partitionValues, takenNames, registered[0]);
+            if (!registered[0]) {
+                // Use the first path to find the partition keys and then use the same partition keys for the rest
+                locationTableBuilder.registerPartitionKeys(partitionKeys);
+                registered[0] = true;
+            }
+            // Use the partition values from each path to build the location table
+            locationTableBuilder.acceptLocation(partitionValues);
+            targetURIs.add(uri);
+        });
+        final Table locationTable = locationTableBuilder.build();
+        buildLocationKeys(locationTable, targetURIs, locationKeyObserver);
+    }
+
+    private void getPartitions(@NotNull final URI relativePath,
+            @NotNull final List<String> partitionKeys,
+            @NotNull final Collection<String> partitionValues,
+            @NotNull final Set<String> takenNames,
+            final boolean registered) {
+        final String relativePathString = relativePath.getPath();
+        int partitioningColumnIndex = 0;
+        // Split the path to get the subdirectory names
+        final String[] subDirs = relativePathString.split(fileSeparator);
+        for (int i = 0; i < subDirs.length - 1; i++) {
+            final String dirName = subDirs[i];
+            if (dirName.isEmpty()) {
+                // Ignore empty directory names
+                continue;
+            }
+            final String[] components = dirName.split("=", 2);
+            if (components.length != 2) {
+                throw new TableDataException("Unexpected directory name format (not key=value) at "
+                        + new File(tableRootDirectory.getPath(), relativePathString));
+            }
+            if (partitioningColumnIndex == maxPartitioningLevels) {
+                throw new TableDataException("Too many partitioning levels at " + relativePathString + ", maximum " +
+                        "expected partitioning levels are " + maxPartitioningLevels);
+            }
+            // We use an empty set to allow duplicate partition keys across files
+            final String columnKey = NameValidator.legalizeColumnName(components[0], Collections.emptySet());
+            if (registered) {
+                // We have already seen another parquet file in the tree, so compare the
```

Comment:
Please remove references to parquet (except examples) in the base class.
```suggestion
                // We have already seen another leaf node in the tree, so compare the
```